capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java

```diff
@@ -128,4 +135,52 @@ public static void createUser(int userId, String userName, String email) throws
     }
     return itemTypes;
   }
+
+  /*
+   * Provided a list of ItemTypes, this method will return a list of Stores
```

Comment:
I'm a little confused with what you're trying to do - you are trying to get all the stores that contain any item on this list, and return the list of stores together with what items they have right? 

I'd do the following: 
1. Get all items from all stores that have the right `itemType`.
2. Group the items by store, and insert the item from each store into a new`Store` object.
3. Return the list of stores.

Reply:
Basically, I am trying to find all the stores that contain _all_ the item types on this list, not just some. That being said, there are many different combinations of item types that exist in a single store.

i.e. I am looking to buy milk and water, and a store has 2 different milks and 2 different waters. There are 4 different combinations of inventory and I want to find those combinations and return them in a list of store objects.

Reply:
okay, how do you plan on displaying the data on the frontend? Do you need all the combinations? I'd argue that it's probably easiest if you just get one item from each itemType (maybe cheapest one) and just return that to start. Your test is also not testing this behavior, because you'll need to mock a store that has at least 2 kinds of 2 different items to test that your combinations work.